From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Sun, 31 Aug 2014 10:52:30 +0200
Subject: FTBFS armel: fixing usage int on ARM audio backend

BZ#1047791

https://bugzilla.mozilla.org/show_bug.cgi?id=1047791

The armel platform uses softfloat for floating numbers. But the audios
backend doesn't support this too. Switching the usage to integer on ARM
based platforms.
Also disable the usage of webm encoding.
---
 mozilla/configure.in |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/mozilla/configure.in b/mozilla/configure.in
index 323a922..b196842 100644
--- a/mozilla/configure.in
+++ b/mozilla/configure.in
@@ -5128,9 +5128,11 @@ AC_SUBST(MOZ_WEBRTC_ASSERT_ALWAYS)
 AC_SUBST(MOZ_SCTP)
 AC_SUBST(MOZ_SRTP)
 
-dnl Use integers over floats for audio on B2G and Android, because audio
-dnl backends for those platforms don't support floats.
-if test "$OS_TARGET" = "Android"; then
+dnl Use integers over floats for audio on B2G and Android
+dnl (regarless of the CPU architecture, because audio
+dnl backends for those platforms don't support floats. We also
+dnl use integers on ARM with other OS, because it's more efficient.
+if test "$OS_TARGET" = "Android" -o "$CPU_ARCH" = "arm"; then
     MOZ_SAMPLE_TYPE_S16=1
     AC_DEFINE(MOZ_SAMPLE_TYPE_S16)
     AC_SUBST(MOZ_SAMPLE_TYPE_S16)
@@ -5524,6 +5526,8 @@ fi
 
 if test -n "$MOZ_TREMOR"; then
     AC_DEFINE(MOZ_TREMOR)
+    # Tremor doesn't have an encoder.
+    MOZ_WEBM_ENCODER=
 fi
 
 if test -n "$MOZ_OPUS"; then
